src/main/webapp/js/map.js

```diff
@@ -87,8 +87,8 @@ function buildInfoWindowInput(lat, lng){
   const button = document.createElement('button');
   button.appendChild(document.createTextNode('Submit'));
   button.onclick = () => {
-    const createdRestaurantID = postMarker(null, lat, lng, textBox.value);
-    createMarkerForDisplay(createdRestaurantID, lat, lng, textBox.value);
+    postMarker(null, lat, lng, textBox.value);
+    createMarkerForDisplay(null, lat, lng, textBox.value);
```

Comment:
Maybe we could have the post request that creates the new restaurant id return the created UUID, and just pass the response back.